trackparser.py

```diff
@@ -45,6 +48,51 @@
 helpNodes = []
 directionDegree = 0
 
+'''
+flip horizontally along x coordinate
+'''
+def flipHorizontally(point, x):
+	newX = 2*x - point.x
+	return Point(newX, point.y)
+
+'''
+rotate point around pivot counter-clockwise
+'''
+def rotatePoint(pivot, point, angle):
+	s = np.sin(np.deg2rad(angle))
+	c = np.cos(np.deg2rad(angle))
+
+	# translate point back to origin:
+	point = Point(
+		point.x - pivot.x,
+		point.y - pivot.y
+	)
+
+	# rotate point
+	xnew = point.x * c - point.y * s
+	ynew = point.x * s + point.y * c
+
+	# translate point back:
+	return Point(
+		xnew + pivot.x,
+		ynew + pivot.y
+	)
+
+
+'''
+get points on an ellipse arc
+'''
+def sampleEllipse(origin, radiusX, radiusY, angle, sampleSize):
+	sampleList = []
+	for step in np.linspace(0, angle, num=sampleSize, endpoint=True):
+		point = Point(
+			origin.x + radiusX * np.cos(np.deg2rad(step)),
+			origin.y + radiusY * np.sin(np.deg2rad(step))
```

Comment:
This is the same formula as previously in line [123](https://github.com/702nADOS/SpeedDreams_SUMO_TrackConverter/pull/4/files#diff-33472c4ed2b616b25235f5d945d3e7f8L123) except it now supports two different radii (radiuses?).
This allows ellipses which are aligned to the x and y axis.